identity-admin/webapp/components/search/Pagination.test.tsx

```diff
@@ -0,0 +1,113 @@
+import Pagination from './Pagination';
+import React from 'react';
+import { render } from '@testing-library/react';
+import { buildSearchUrl } from '../../utils/search-util';
+
+const renderComponent = (currentPage: number, pageCount: number) =>
+  render(<Pagination currentPage={currentPage} pageCount={pageCount} />);
+const getFirstLink = (container: HTMLElement) =>
+  container.querySelector(
+    '.user-pagination__item.user-pagination__item--first'
+  );
+const getLastLink = (container: HTMLElement) =>
+  container.querySelector('.user-pagination__item.user-pagination__item--last');
+const getAllLinks = (container: HTMLElement) =>
+  container.getElementsByClassName('user-pagination__item');
+const expectedLinkUrl = (expectedPage: string) =>
```

Comment:
Are we just duplicating functionality here? Would it be better in this case to assert on hard-coded strings?

Reply:
The main reason for not using hard coded strings in each assertion was that they were long and mostly the same, with the only parameter that varies between the tests being in the middle of the URL:

```
/?email=bob%40email.com&name=Bob&page=5&sort=email&sortDir=1&status=active
```

I don't want to hard code those strings in each test because I think it makes the tests less readable, but I'd be happy to change this function to have the hard coded string:

```
const expectedLinkUrl = (expectedPage: string) =>
  '/?email=bob%40email.com&name=Bob&page=' +
  expectedPage +
  '&sort=email&sortDir=1&status=active';
```

Reply:
Yeah, that's a good solution here.

But if all we care about is the `page=5` bit, maybe that's all we should test for:

```ts
expect(next).toHaveAttribute('href', expect.stringContaining('page=5');
```

Up to you, I'm happy either way.

Reply:
It's not so much that we don't care about the other params - we do. It's just that I want to make it clear in the tests that that's the only one that should vary between each scenario